modules/logging/filters.go

```diff
@@ -100,12 +101,15 @@ func (f *HashFilter) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
 // Filter filters the input field with the replacement value.
 func (f *HashFilter) Filter(in zapcore.Field) zapcore.Field {
 	if array, ok := in.Interface.(caddyhttp.LoggableStringArray); ok {
-		for i, s := range array {
-			array[i] = hash(s)
+		newArray := slices.Clone(array)
+		for i, s := range newArray {
+			newArray[i] = hash(s)
 		}
+		in.Interface = newArray
```

Comment:
To be tested, but there is probably no need to copy the existing data:

```suggestion
		newArray := make(caddyhttp.LoggableStringArray, len(array)
		for i, s := range array {
			newArray[i] = hash(s)
		}
		in.Interface = newArray
```

Reply:
Thanks, I was going to suggest the same thing.